tests/services/user.service.spec.ts

```diff
@@ -0,0 +1,51 @@
+import { jsonData, USER_JUNWEI, USER_Q, USER_SHUMING } from '../constants/data.constants'
+import { UserService } from '../../src/app/core/services/user.service'
+import { User, UserRole } from '../../src/app/core/models/user.model';
+import { of } from 'rxjs';
+
+let dataService: any;
+
+describe('UserService', () => {
+  describe('.createUserModel(loginId)', () => {
+    beforeAll(() => {
+      dataService = jasmine.createSpyObj("DataService", ['getDataFile']);
+      dataService.getDataFile.and.returnValue(of(jsonData));
+    });
+
+    it('creates a Student user correctly', async () => {
+      await createAndVerifyUser(USER_JUNWEI.loginId, USER_JUNWEI);
+    });
+
+    it('creates a Tutor user correctly when loginId is very short', async () => {

```

Comment:
I've modified this test case to use a single-character loginId, because no other test was checking this type of edge case.